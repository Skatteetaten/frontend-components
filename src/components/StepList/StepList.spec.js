import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { StepList, Step } from '../index';

function oppsettShallow(props) {
  return shallow(
    <StepList>
      <Step {...props} />
    </StepList>
  );
}

function oppsettMount(props) {
  return mount(
    <StepList>
      <Step {...props} />
    </StepList>
  );
}

describe('StepList komponent', () => {
  it('matcher snapshot', () => {
    const wrapper = oppsettMount();
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('rendrer steplist uten at den krÃ¦sjer', () => {
    oppsettShallow({});
  });

  it('rendrer steplist med riktig props', () => {
    const wrapper = oppsettShallow({
      stepTitle: 'Jobber du?',
      stepId: 'step-1-1',
    });

    expect(wrapper.find('Step').prop('stepTitle')).toEqual('Jobber du?');
    expect(wrapper.find('Step').prop('stepId')).toEqual('step-1-1');
  });

  it('rendrer riktig html for for steg type "result"', () => {
    oppsettShallow({});

    const wrapper = oppsettMount({
      stepType: 'result',
    });

    expect(wrapper.find('Step').prop('stepType')).toEqual('result');
    expect(wrapper.find('Icon').length).toBe(1);
    expect(wrapper.exists('.stepLine')).toEqual(false);
    expect(wrapper.containsMatchingElement(<hr />)).toEqual(false);
  });

  it('rendrer riktig html for steg type "next"', () => {
    oppsettShallow({});

    const wrapper = oppsettMount({
      stepType: 'next',
    });

    expect(wrapper.exists('.stepNumber')).toEqual(false);
  });
});
