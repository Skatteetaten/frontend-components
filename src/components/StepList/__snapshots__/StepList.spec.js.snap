// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepList komponent matcher snapshot 1`] = `
<StepList>
  <div
    className="stepList-53"
    role="region"
  >
    <Grid
      padding="0"
    >
      <div
        className="ske-Grid-58 "
        dir="ltr"
      >
        <Step
          key=".0"
          stepNumber={1}
        >
          <div
            className="wrapperStep-65"
            key="1"
          >
            <GridRow
              rowSpacing="0px"
            >
              <div
                className="ske-Grid-row-62 "
              >
                <GridCol
                  noSpacing={true}
                >
                  <div
                    className="ske-Grid-col-63 "
                  >
                    <GridRow
                      rowSpacing="0px"
                    >
                      <div
                        className="ske-Grid-row-62 "
                      >
                        <GridCol
                          lg={10}
                          sm={12}
                          xxxl={8}
                        >
                          <div
                            className="ske-Grid-col-64 "
                          >
                            <span
                              className="stepLine-51"
                            />
                            <span
                              className=""
                            />
                            <NumberIcon
                              styles={
                                Object {
                                  "arrowLine": "arrowLine-42",
                                  "containerStep": "containerStep-43",
                                  "content": "content-44",
                                  "divider": "divider-45",
                                  "nextStep": "nextStep-46",
                                  "numberWrapper": "numberWrapper-47",
                                  "stepAction": "stepAction-48",
                                  "stepContent": "stepContent-49",
                                  "stepContentInner": "stepContentInner-50",
                                  "stepLine": "stepLine-51",
                                  "stepLineTop": "stepLineTop-52",
                                  "stepList": "stepList-53",
                                  "stepNumber": "stepNumber-54",
                                  "subComponentStyles": Object {},
                                  "title": "title-55",
                                  "titleText": "titleText-56",
                                  "wrapperStep": "wrapperStep-65",
                                }
                              }
                            >
                              <div
                                className="numberWrapper-47"
                              >
                                <div
                                  className="stepNumber-54"
                                >
                                  <span />
                                </div>
                              </div>
                            </NumberIcon>
                            <hr
                              className="divider-45"
                            />
                            <div>
                              <div
                                className="stepContent-49"
                              >
                                <div
                                  className="stepContentInner-50"
                                />
                              </div>
                            </div>
                          </div>
                        </GridCol>
                      </div>
                    </GridRow>
                  </div>
                </GridCol>
              </div>
            </GridRow>
          </div>
        </Step>
      </div>
    </Grid>
  </div>
</StepList>
`;
